src/LibHdb++.h

```diff
@@ -1,18 +1,27 @@
-//=============================================================================
-//
- // file :        LibHdb++.h
-//
-// description : Include for the LibHdb library.
-//
-// Author: Graziano Scalamera
-//
-//
-//=============================================================================
-
-#ifndef _LIB_HDB_H
-#define _LIB_HDB_H
-
-#include <tango.h>
+/* Copright (C) 2014-2017
+   Elettra - Sincrotrone Trieste S.C.p.A.
+   Strada Statale 14 - km 163,5 in AREA Science Park
+   34149 Basovizza, Trieste, Italy.
+
+   This file is part of libhdb++.
+
+   libhdb++cassandra is free software: you can redistribute it and/or modify
+   it under the terms of the Lesser GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   libhdb++cassandra is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the Lesser
+   GNU General Public License for more details.
+
+   You should have received a copy of the Lesser GNU General Public License
+   along with libhdb++cassandra.  If not, see <http://www.gnu.org/licenses/>. */
+
+#ifndef _HDBPP_H
+#define _HDBPP_H
+
+#include <tango/tango.h>
```

Comment:
this has to be #include <tango.h>

Reply:
Actually, the offical releases install tango into /usr/include/tango. This is also the standard pattern to manage headers in a linux distrubtion (putting everything in include invites conflicts).

Changing means we can not build against the standard releases without for ever supplying additional flags. This looks pretty bad for new users.

So, I normalised this, and now anyone with a non-standard install can use a single additional include the flags to continue building. 

Further, the latest tango is putting all headers under /usr/include/tango, so we will need to move to this anyway.

We do currently have bit of a mixed install by different packages. While tango installs to /usr/include/tango, it tries to include tango headers from /usr/include (this appears to be a problem with the build). To work around this, we currently supply a path. Next tango release this all goes away, and we can simply run make without any additinal flags or settings!

Reply:
Also.. Since we are going to work towards inclusion in debian, our headers will have to follow the standard linux pattern. Otherwise the packages will be rejected.

Reply:
when installing tango from sources headers are installed in: <install_prefix>/include/tango, if you do not include this full path but just <install_prefix>/include then tango.h cannot find other tango includes (e.g. tango_config.h). You should edit tango sources to have #include <tango/tango_config.h> in tango.h for example.

Reply:
Yes, I agree totally. And, lucky for us, this has happened in the latest Tango code in github (for tango 10). The include paths have been fixed. I spent a lot of time trying to work out what was happening with different tango distrubutions (they all do header installs slightly different). 

Dont worry to much about my changes above. They work perfectly with future tango, and support the current release via a single additional include path passed to the build (TANGO_INC, for makefules and DCMAKE_INCLUDE_PATH in cmake). If you take a look in the debian/rules file, you can see how its being used with DCMAKE_INCLUDE_PATH in this case.

Any other non-standard paths can also be passed to the build via these flags. So we keep compatibility, yet on a standard install system, we compile without having to do additional setup!

Reply:
So, until the fix on tango, both <tango_install_prefix>/include/tango and <tango_install_prefix>/include needs to be passed to CMAKE_INCLUDE_PATH

Reply:
Yes, or just one when you install to standard locations. Let me expand on the rational here a little, since I feel it's important.

Most of the HDB++ modules did not build here, due to our non standard tango install. Between here and there, it was incompatible to a degree (we installed all tango headers to prefix/include). Worse, different ways of installing tango could result in different header installs.

So, I changed the build on each module with two aims.

1) Support a standard setup by default. New users can get the code, and build it with ease if they are not installing to unusual locations. This also means we can package and work with docker more easily. On tango 10, with a standard install, you can simply type make.

2) Continue to support non-standard locations, but not at the expense of a simple standard build. Basically, you still have the flags, and still need the pass them like before, but the method may have changed a little. In cmakes case, we follow standard usage, with the Makefiles, allow environment variables to pass in includes like before, but now we only set the ones we need.

Trying to support multiple unique installs efficiently gets difficult over time. 

To support the changes, I added plenty of documentation to each project. 

For the device servers I did not change the include path, since I believe pogo maintains that? I am not sure how all this will work in future.

Reply:
I have been following this up.

I believe the intention of the cppTango guys is to install tango.h to /usr/include, and install the other headers to /usr/include/tango.

With this, I am reverting this change. If things in cppTango change, then we can follow it at a late date.